tests/integration/search_test.t

```diff
@@ -0,0 +1,167 @@
+		ingredients_text => "apple, milk, eggs",
+		origin => "france",
+		serving_size => "10g",
+		packaging_text => "no",
+		action => "process",
+		type => "add",
+		".submit" => "submit"
+	},
+	{
+		code => '200000000039',
+		lang => "es",
+		product_name => "Vegan Test Snack",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "apple, water, palm oil",
+		origin => "spain",
+		serving_size => "5g",
+		packaging_text => "no",
+		action => "process",
+		categories => "snacks",
+		type => "add",
+		".submit" => "submit"
+	},
+	{
+		code => '200000000045',
+		lang => "es",
+		product_name => "Vegan Test Snack",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "apple, water",
+		origin => "France",
+		serving_size => "5g",
+		packaging_text => "no",
+		action => "process",
+		categories => "breakfast cereals",
+		type => "add",
+		".submit" => "submit"
+	}
+);
+
+remove_all_users();
+
+remove_all_products();
+
+wait_dynamic_front();
+
+my $ua = new_client();
+
+my %create_user_args = (%default_user_form, (email => 'bob@gmail.com'));
+create_user($ua, \%create_user_args);
+
+foreach my $ref (@products) {
+	create_product($ua, $ref);
+}
+
+my @tests = (
+	["q1", construct_test_url("/cgi/search.pl?action=process&json=1", "world")],
+	["q2", construct_test_url("/cgi/search.pl?action=process&json=1&ingredients_from_palm_oil=without", "world")],
+	[
+		"q3",
+		construct_test_url(
+			"/cgi/search.pl?action=process&code=200000000034,200000000039&fields=code,product_name&json=1", "world"
+		)
+	],
+	[
+		"q4",
+		construct_test_url(
+			"/cgi/search.pl?action=process&tagtype_0=categories&tag_contains_0=contains&tag_0=breakfast_cereals&tagtype_1=nutrition_grades&tag_contains_1=contains&ingredients_from_palm_oil=without&json=1",
+			"world"
+		)
+	]
+);
+
+## TODO : find out whats up with these quesries below :(
+# my $query_3_json = get("http://world.openfoodfacts.localhost/api/v2/search?code=200000000039,200000000038&fields=code,product_name");
+# my $query_3_json_decoded = decode_json($query_3_json);
+
+my $update_expected_results = 1;
+
+if ((defined $update_expected_results) and (!-e $expected_dir)) {
+	mkdir($expected_dir, 0755) or die("Could not create $expected_dir directory: $!\n");
+}
+
+foreach my $test_ref (@tests) {
+	my $testid = $test_ref->[0];
+	my $query_ref = $test_ref->[1];
+	my $json = get($query_ref);
+
+	my $decoded_json = decode_json($json);
+
+	my $length = @{$decoded_json->{'products'}};
+	my $count;
+	for ($count = 0; $count < $length; $count++) {
+		foreach my $field ("created_t", "last_modified_t") {
+			delete($decoded_json->{'products'}[$count]{created_t});
+		}
+	}
+	is(compare_to_expected_results($decoded_json, "$expected_dir/$testid.json", $update_expected_results), 1);
```

Comment:
We need to call the functions to remove dates from fields, if I run the tests today, I get different values from the ones you had yesterday:

```
          "entry_dates_tags" : [
-            "2022-09-04",
+            "2022-09-05",
             "2022-09",
             "2022"
          ],
@@ -495,13 +495,13 @@
             "en:1"
          ],
          "last_edit_dates_tags" : [
-            "2022-09-04",
+            "2022-09-05",
             "2022-09",
             "2022"
          ],
          "last_editor" : "tests",
          "last_modified_by" : "tests",
-         "last_modified_t" : 1662334632,
+         "last_modified_t" : 1662364495,

```

Reply:
I'll add these fields to the loop where we are sanitizing the hash